### ISSUE TITLE
configured users_excluded attribute in okta_group_rule resource is not getting saved to state

### ISSUE THREAD

**Issue by dlethin** [okta/terraform-provider-okta#1018]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
Terraform v1.1.7
on darwin_amd64
+ provider registry.terraform.io/okta/okta v3.22.1

```
### Affected Resource(s)

- okta_group_rule

### Terraform Configuration Files

```hcl
locals = {
  role_map = {
    "dev-admin" = [
      "[REACTED_GROUP_ID_1]",
      "[REACTED_GROUP_ID_2]",
    ]
  }
}
resource "okta_group_rule" "mgmt_group" {
  for_each = local.role_map

  name              = "z-aws-${each.key}-rule"
  status            = "ACTIVE"

  group_assignments = each.value

  remove_assigned_users = true
  users_excluded = []
  expression_type   = "urn:okta:expression:1.0"
  expression_value  = "isMemberOfGroupName(\"${okta_group.aws_mgmt_group[each.key].name}\")"
}

```

### Expected Behavior

After applying plan, subsequent plans should show any changes made outside of terraform

### Actual Behavior

Receiving a warning indicating changes have been made outside terraform for the `users_excluded` field:

```
Note: Objects have changed outside of Terraform

Terraform detected the following changes made outside of Terraform since the
last "terraform apply":

  # okta_group_rule.mgmt_group["dev-admin"] has changed
  ~ resource "okta_group_rule" "mgmt_group" {
        id                    = "[REDACTED]"
        name                  = "z-aws-dev-admin-rule"
      + users_excluded        = []
        # (5 unchanged attributes hidden)
    }


Unless you have made equivalent changes to your configuration, or ignored the
relevant attributes using ignore_changes, the following plan may include
actions to undo or respond to these changes.

─────────────────────────────────────────────────────────────────────────────

No changes. Your infrastructure matches the configuration.

Your configuration already matches the changes detected above. If you'd like
to update the Terraform state to match, create and apply a refresh-only plan:
  terraform apply -refresh-only
```

As you can see in the resource configuration, we are explicitly setting `users_excluded` to an empty list, but it seems to be getting set instead to null as seen in the state file:

```
{
  "version": 4,
  "terraform_version": "1.1.7",
  "serial": 24,
  "lineage": "b7a4a891-b177-1138-2ae5-91a30eee6a49",
  "outputs": {},
  "resources": [
    {
      "mode": "managed",
      "type": "okta_group_rule",
      "name": "mgmt_group",
      "provider": "provider[\"registry.terraform.io/okta/okta\"]",
      "instances": [
        {
          "index_key": "dev-admin",
          "schema_version": 0,
          "attributes": {
            "expression_type": "urn:okta:expression:1.0",
            "expression_value": "isMemberOfGroupName(\"z-aws-mgmt-dev-admin\")",
            "group_assignments": [
              "[REACTED_GROUP_ID_1]",
              "[REACTED_GROUP_ID_2]",
            ],
            "id": "[REACTED]",
            "name": "z-aws-dev-admin-rule",
            "remove_assigned_users": true,
            "status": "ACTIVE",
            "users_excluded": null
          },
          "sensitive_attributes": [],
          "private": "[REACTED]=",
          "dependencies": [
            "okta_group.aws_mgmt_group"
          ]
        },
      ]
    }
  ]
}
```

thanks.

**Comment by monde** [comment]
Thanks @dlethin I will look into this.

**Comment by monde** [comment]
I believe this is resolved, ref #1016
